Xamarin.Forms.Platform.UAP/TabbedPageRenderer.cs

```diff
@@ -333,5 +352,80 @@ void UpdateToolbarPlacement()
+
+        void UpdateBarIcons()
+        {
+            if (Control == null)
+                return;
+
+            if (Element.IsSet(Specifics.HeaderIconsEnabledProperty))
+            {
+                bool headerIconsEnabled = Element.OnThisPlatform().GetHeaderIconsEnabled();
+                bool invalidateMeasure = false;
+
+                // Get all stack panels affected by update.
+                var stackPanels = Control.GetDescendantsByName<WStackPanel>(TabBarHeaderStackPanelName);
+                foreach (var stackPanel in stackPanels)
+                {
+                    foreach (var stackPanelItem in stackPanel.Children)
+                    {
+                        if (stackPanelItem is WImage)
+                        {
+                            // Update icon image.
+                            var tabBarImage = stackPanelItem as WImage;
+                            if (tabBarImage.GetValue(FrameworkElement.NameProperty).ToString() == TabBarHeaderImageName)
+                            {
+                                if (headerIconsEnabled)
+                                {
+                                    if (Element.IsSet(Specifics.HeaderIconsSizeProperty))
+                                    {
+                                        Size iconSize = Element.OnThisPlatform().GetHeaderIconsSize();
+                                        tabBarImage.Height = iconSize.Height;
+                                        tabBarImage.Width = iconSize.Width;
+                                    }
+                                    tabBarImage.HorizontalAlignment = WHorizontalAlignment.Center;
+                                    tabBarImage.Visibility = WVisibility.Visible;
+                                }
+                                else
+                                {
+                                    tabBarImage.Visibility = WVisibility.Collapsed;
+                                }
+
+                                invalidateMeasure = true;
+                            }
+                        }
+                        else if (stackPanelItem is WTextBlock)
+                        {
+                            // Update text block.
+                            var tabBarTextblock = stackPanelItem as WTextBlock;
+                            if (tabBarTextblock.GetValue(FrameworkElement.NameProperty).ToString() == TabBarHeaderTextBlockName)
+                            {
+                                if (headerIconsEnabled)
+                                {
+                                    // Remember old values so we can restore them if icons are collapsed.
+                                    // NOTE, since all Textblock instances in this stack panel comes from the same
+                                    // style, we just keep one copy of the value (since they should be identical).
+                                    _oldBarTextBlockTextAlignment = tabBarTextblock.TextAlignment;
```

Comment:
One thing I noticed here when messing with your sample is that initially these two *old* values are 
Left and Stretch

Then when you set the size to something larger like (170,170) for the icons they become
Center and Center

And now they are just Center and Center forever and they won't go back to Stretch and Left

You don't really see the result of this because the surrounding stackpanel constrains the elements so these alignments don't really have any effect.  To see the effect if you set the containing stackpanel to a larger width
```
<StackPanel Orientation="Vertical" Name="TabbedPageHeaderStackPanel" Background="Pink" Width="100">
```

Then do
SetSize to some non larger non default (170,170) => Set size back to default => hide icons 
You'll now see the text is centered whereas initially it was left aligned

It's a bit edge case but I wanted to mention it and see what you think

Reply:
Will investigate this edge case, thanks for spotting!